### PR TITLE
Move/createmoduleextension

### PR DESCRIPTION
**Pull request by ErichBSchulz**
wow this took me hours and hours 

but doesnt look like much

there are some issues with Sean's tool

I gotta go on hot date with spouse so creating a PR with what I've done so far

I hope its a step forward!

### CODE REVIEW

**Comment by seanmadsen** [comment]
> there are some issues with Sean's tool

Would love it if you could [report those issues](https://github.com/seanmadsen/webpage2md/issues)

File: docs/extensions/civix.md [hunk 1]
```diff
@@ -1 +1,582 @@
-Coming soon!
+# Civix
+
+>> ![information](../img/info.png)
```
Comment (seanmadsen) [inline docs/extensions/civix.md]: I like these *information* and *danger* blocks. Great idea! 

As an aside: I wonder if (in the future) we might be able to find a way to make this easier. I'm imagining something like 

```
{information
I am an informational message.
}
```

(Not a project that I feel should be investigated before merging this PR, but just a random thought.)

File: docs/extensions/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Troubleshooting
```
Comment (seanmadsen) [inline docs/extensions/troubleshooting.md]: :heart: the idea for this page!

**Comment by seanmadsen** [comment]
FYI @ErichBSchulz  I like what you're doing overall here! I've skimmed it over, but I don't quite feel qualified to offer an official review of these changes because I don't know the content well enough. Will defer to others.

**Comment by ErichBSchulz** [comment]
I have kept changes to narrative and copy edits. I am still finishing bringing in the key bits from the command line blocks that the tool left behind. (some progress this morning)

**Comment by ErichBSchulz** [comment]
I think this is ready for merging

there is still some tidying and clarifying to go but I think if we merge we can then deal with those as PRs and hopefully get some input from others in a better position to clarify

File: docs/extensions/civix.md [hunk 3]
```diff
@@ -1 +1,636 @@
...
+
+There are many different features that you can add to a module-extension
+at your discretion. A few possibilities:
+
+### Add a basic web page
+
+CiviCRM uses a typical web-MVC architecture. To implement a basic web
+page, you must create a PHP controller class, create a Smarty template
+file, and create a routing rule. You can create the appropriate files by
+calling "civix generate:page"
```
Comment (totten) [inline docs/extensions/civix.md]: For inline references to code/commands/symbols, you can use backtick. Compare: "civix generate:page" vs `civix generate:page`

**Comment by totten** [comment]
Looks like good progress. :)

There's a few more places where backticks would be better than double-quotes but don't need to block on that.

**Pull request merged**